test/language/import/import-defer/errors/module-throws/third-party-evaluation-after-defer-import.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2024 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-module-namespace-exotic-objects-get-p-receiver-EnsureDeferredNamespaceEvaluation
+description: >
+  Module evaluation errors are thrown
+info: |
+  [[Get]] ( _P_, _Receiver_ )
+    1. ...
+    1. If _O_.[[Deferred]] is **true**, perform ? EnsureDeferredNamespaceEvaluation(_O_).
+    1. ...
+
+  EnsureDeferredNamespaceEvaluation( _O_ )
+    1. ...
+    1. Perform ? EvaluateSync(_m_).
+    1. ...
+
+  EvaluateSync ( _module_ )
+    1. ...
+    1. Let _promise_ be ! _module_.Evaluate().
+    1. Assert: _promise_.[[PromiseState]] is either ~fulfilled~ or ~rejected~.
+    1. If _promise_.[[PromiseState]] is ~rejected~, then
+      1. Return ThrowCompletion(_promise_.[[PromiseResult]]).
+    1. ...
+
+flags: [module]
+features: [import-defer, top-level-await]
+---*/
+
+import defer * as ns from "./throws_FIXTURE.js";
+
+await import("./throws_FIXTURE.js").catch(() => {});
```

Comment:
For my own understanding, the URIError should be thrown regardless of whether this line is present, right?

Or maybe not, since the testing plan says "import and import defer + exec only execute the module once"

In any case, you should probably have both a `.then` and a `.catch` callback, and assert that only one is executed.

Reply:
This test for the following scenario:
- you get a deferred namespace of a module that has not been evaluated yet
- somebody else (in this case the dynamic import, but in the real world it will probably be a static import somewhere in another file) evaluates it. The test doesn't care about the details of this evaluation, just that it happens
- you try to access the deferred namespace you originally got a hold of, and it re-throws the error cached by the other evaluation

In the plan, it's "Property access re-throws the evaluation error > On first access for already evaluated modules"